lib/proxyHandler.js

```diff
@@ -80,10 +80,49 @@ handler.prototype.parse_req = function(req, parentReq, res) {
     }
 };
 
-handler.prototype.eval_request = function(spec) {
-    return function (req) {
-        return this.parse_req(spec, req);
+function _assemble(spec) {
+    var traverse = function(path) {
+        // A simple dotted path reference
+        var context = "req"
+        path.split(/\./).forEach(function(pathBit, index) {
```

Comment:
We could actually require that identifiers conform to `[a-zA-Z_]+`. That would let us simply validate the path, and skip the rewriting.